crates/bevy_color/src/srgba.rs

```diff
@@ -371,47 +373,13 @@ pub enum HexColorError {
     Char(char),
 }
 
-/// All color channels are scaled directly,
-/// but alpha is unchanged.
-///
-/// Values are not clamped.
-impl Mul<f32> for Srgba {
-    type Output = Self;
-
-    fn mul(self, rhs: f32) -> Self {
-        Self {
-            red: self.red * rhs,
-            green: self.green * rhs,
-            blue: self.blue * rhs,
-            alpha: self.alpha,
-        }
-    }
-}
-
-impl Mul<Srgba> for f32 {
-    type Output = Srgba;
-
-    fn mul(self, rhs: Srgba) -> Srgba {
-        rhs * self
-    }
-}
-
-/// All color channels are scaled directly,
-/// but alpha is unchanged.
-///
-/// Values are not clamped.
-impl Div<f32> for Srgba {
-    type Output = Self;
+impl_color_add!(Srgba, [red, green, blue]);
+impl_color_sub!(Srgba, [red, green, blue]);
+impl_color_mul!(Srgba, [red, green, blue]);
+impl_color_div!(Srgba, [red, green, blue]);
+impl_color_neg!(Srgba, [red, green, blue]);
```

Comment:
I'm not sure about this. Addition in sRGB is not well defined because it's a non-linear colorspace. It's one of these things that are footgunny but users expect because of a surface-level understanding of color.

I would prefer not adding this, and instead documenting that to perform mathematical operations on color, they first convert to `LinearSrgba`, then convert back at the end of the computation.

It's definitely a correctness above performance issue, as it requires performing `sqrt` and `pow` operations during the conversion and back, but I think it is worth it to guide users who might not know better.

Reply:
I think it is fine, as the documentation of what that addition does is quite extensive. These operators are explicitly componentwise and (to me at least) adding two sRGB colors in linear RGB space would be much more unintuitive allthough it might be more correct.

Reply:
I disagree that we should provide escape hatches based on unintuitiveness. Color is hard, and there are lots of misconceptions, and to me it's all the more reason to make it easy to do things correctly, and hard to do them incorrectly. In most cases operations on gamma-corrected sRGB is wrong and only done due to lack of understanding, where it would have made more sense to store and operate on linear colors in the first place.

I also disagree with the documentation claim as well, there is no mention of addition at all, and in particular warnings with operating on gamma-corrected sRGB values. These should, at a minimum, be added to the documentation if we are to provide high-quality documentation and allow operations on gamma-coorected sRGB values.

Reply:
Yes, color is hard, but we have agreed above, that we are treating the color channels as a Vector in all cases. As outlined above, there are usecases for this and the documentation for addition, subtraction, etc. is available both at [the crate level](https://github.com/bevyengine/bevy/pull/12460/files#diff-753e998821c359c48ed30e5028a572ab257bd5d0b4f467a37a073eb8778cf790R64-R80) and at [each implementation](https://github.com/bevyengine/bevy/pull/12460/files#diff-753e998821c359c48ed30e5028a572ab257bd5d0b4f467a37a073eb8778cf790R200-R212) of `Add`, `AddAssign`, `Sub`, `SubAssign`, etc.

Reply:
I'd have liked more explicit documentation about why doing component-wise addition on gamma-corrected sRGB values might not be what they want. But it seems nobody else is raising objections to this, so I guess this is fine.